mongodbatlas/resource_mongodbatlas_project_ip_whitelist.go

```diff
@@ -25,226 +28,209 @@ func resourceMongoDBAtlasProjectIPWhitelist() *schema.Resource {
 		},
 		Schema: map[string]*schema.Schema{
 			"project_id": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
-			"whitelist": {
-				Type:     schema.TypeSet,
-				Required: true,
-				ForceNew: true,
-				Set:      filterParamsHash,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"cidr_block": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Computed: true,
-							ForceNew: true,
-							ValidateFunc: func(i interface{}, k string) (s []string, es []error) {
-								v, ok := i.(string)
-								if !ok {
-									es = append(es, fmt.Errorf("expected type of %s to be string", k))
-									return
-								}
-
-								_, ipnet, err := net.ParseCIDR(v)
-								if err != nil {
-									es = append(es, fmt.Errorf(
-										"expected %s to contain a valid CIDR, got: %s with err: %s", k, v, err))
-									return
-								}
-
-								if ipnet == nil || v != ipnet.String() {
-									es = append(es, fmt.Errorf(
-										"expected %s to contain a valid network CIDR, expected %s, got %s",
-										k, ipnet, v))
-									return
-								}
-								return
-							},
-						},
-						"ip_address": {
-							Type:         schema.TypeString,
-							ForceNew:     true,
-							Optional:     true,
-							Computed:     true,
-							ValidateFunc: validation.SingleIP(),
-						},
-						"comment": {
-							Type:         schema.TypeString,
-							Optional:     true,
-							Computed:     true,
-							ValidateFunc: validation.NoZeroValues,
-						},
-					},
+			"cidr_block": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"aws_security_group", "ip_address"},
+				ValidateFunc: func(i interface{}, k string) (s []string, es []error) {
+					v, ok := i.(string)
+					if !ok {
+						es = append(es, fmt.Errorf("expected type of %s to be string", k))
+						return
+					}
+
+					_, ipnet, err := net.ParseCIDR(v)
+					if err != nil {
+						es = append(es, fmt.Errorf("expected %s to contain a valid CIDR, got: %s with err: %s", k, v, err))
+						return
+					}
+
+					if ipnet == nil || v != ipnet.String() {
+						es = append(es, fmt.Errorf("expected %s to contain a valid network CIDR, expected %s, got %s", k, ipnet, v))
+						return
+					}
+					return
 				},
 			},
+			"ip_address": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"aws_security_group", "cidr_block"},
+				ValidateFunc:  validation.SingleIP(),
+			},
+			"aws_security_group": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"ip_address", "cidr_block"},
+			},
+			"comment": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
```

Comment:
I tried to achieve that, but like we can create multiple whitelist entries at the same time, the way to do that doesn't allow us to update the comments. Let me explain how to works when you want to create multiple whitelist entries. Imagine you want to create ten whitelist entries, so you apply in Terraform to sent the ten requests (note that the MongoDB server doesn't support multiple requests at the same time). The MongoDB server will return multiple errors (internal error 500 code) when this happens; the resource will send the requests again until the server accepts them. So this turns complicated because if we want to update the comment, we need to handle it like this, but extra more complication due to we have an updating function, and it needs to handle the state or the other attributes, it becomes unstable. For this reason I decided to remove.